src/modules/views/AgentDetails.jsx

```diff
@@ -120,7 +120,7 @@ const AgentDetails = (props) => {
   }
 
   const determinePath = () => {
-    return !state.currentUser ? history.push("/welcome") : history.push("/mission-control") 
+    return !state.currentUser ? history.push("/") : history.push("/mission-control") 
```

Comment:
I think we had the same idea here - nice call. I added it into a useEffect() so that it automatically sends to the welcome page if there isn't a currentUser in state, but looks like with the logic you've added it'll be unnecessary. Nice work!